### PR TITLE
JAMES-1982 Can not use CLI for configuring quota with Guice

### PR DESCRIPTION
**Pull request by chibenwa**
- Bind the relevant MBeans to enable all commands
 - Add a probe for quotas (the missing one) for when we

### CODE REVIEW

File: server/container/cli/src/main/java/org/apache/james/cli/ServerCmd.java [hunk 1]
```diff
@@ -86,24 +86,8 @@ private static Options createOptions() {
...
-                    new JmxQuotaProbe().connect(jmxConnection),
-                    new JmxSieveProbe().connect(jmxConnection)
-                )
-                .executeCommandLine(cmd);
-            stopWatch.split();
-            print(new String[] { Joiner.on(' ')
-                    .join(cmdType.getCommand(), "command executed sucessfully in", stopWatch.getSplitTime(), "ms.")},
-                System.out);
-            stopWatch.stop();
+            doMain(args);
```
Comment (chibenwa) [inline server/container/cli/src/main/java/org/apache/james/cli/ServerCmd.java]: Extraction is done here to ease testability. 

It is used later.

**Comment by chibenwa** [comment]
Added corrections for CLI with Guice:
 - NPE when using basic mailboxes operations
 - NPE with quotas
 - Stopping JMX server
 - Corretion of doubled CLI output

With that CLI should be ready for release :trollface:

**Comment by chibenwa** [comment]
```

[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 11.191 sec <<< FAILURE! - in org.apache.james.mpt.imapmailbox.elasticsearch.ElasticSearchMailboxTest
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] testSearchAtomsUS(org.apache.james.mpt.imapmailbox.suite.UidSearchOnIndex)  Time elapsed: 10.737 sec  <<< ERROR!
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] org.apache.james.mpt.protocol.ProtocolSession$InvalidServerResponseException: 
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] 
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] Location: /org/apache/james/imap/scripts/UidSearchAtomsIndexer.test:2322
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] LastClientMsg: A203 UID SEARCH FROM "Timothy"
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] Expected: '\* SEARCH 1'
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] Actual   : '* SEARCH'
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] 	at org.apache.james.mpt.imapmailbox.suite.UidSearchOnIndex.testSearchAtomsUS(UidSearchOnIndex.java:40)
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] 
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] 
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] Results :
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] 
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e] Tests in error: 
[Building Java 8 james-1982-f58c8293-ea30-478a-bfec-22a7a2ce857e]   UidSearchOnIndex.testSearchAtomsUS:40->AbstractSimpleScriptedTestProtocol.scriptTest:78->AbstractProtocolTestFramework.runSessions:104 » InvalidServerResponse
```

test this please

File: server/container/guice/jmx/src/main/java/org/apache/james/modules/server/JMXServer.java [hunk 2]
```diff
@@ -117,7 +119,8 @@ private void doStop() {
             });
             registeredKeys.clear();
             jmxConnectorServer.stop();
-        } catch (IOException e) {
+            restrictingRMISocketFactory.getSockets().forEach(Throwing.consumer(ServerSocket::close).sneakyThrow());
```
Comment (rouazana) [inline server/container/guice/jmx/src/main/java/org/apache/james/modules/server/JMXServer.java]: indent

File: server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.google.inject.Inject;
+import com.google.inject.multibindings.Multibinder;
+
+public class DataCommandsIntegrationTest {
+
+    public static final String DOMAIN_COM = "domain.com";
```
Comment (rouazana) [inline server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java]: DOMAIN?

File: server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    @Test
+    public void removeAddressMappingShouldWork() throws Exception {
+        String redirectionAddress = "redirect@apache.org";
+        ServerCmd.doMain(new String[] {"-h", "127.0.0.1", "-p", "9999", "addaddressmapping", USER, DOMAIN_COM, redirectionAddress});
+
+        ServerCmd.doMain(new String[] {"-h", "127.0.0.1", "-p", "9999", "removeaddressmapping", USER, DOMAIN_COM, redirectionAddress});
+
+        assertThat(dataProbe.listMappings())
+            .isNull();
```
Comment (rouazana) [inline server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java]: null? should not be empty instead?

File: server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    @Test
+    public void removeRegexMappingShouldWork() throws Exception {
+        String regex = "regex";
+        ServerCmd.doMain(new String[] {"-h", "127.0.0.1", "-p", "9999", "addregexmapping", USER, DOMAIN_COM, regex});
+
+        ServerCmd.doMain(new String[] {"-h", "127.0.0.1", "-p", "9999", "removeregexmapping", USER, DOMAIN_COM, regex});
+
+        assertThat(dataProbe.listMappings())
+            .isNull();
```
Comment (rouazana) [inline server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java]: idem?

File: server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java (hunk 4 above)
Comment (chibenwa) [inline server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java]: This PR does not aim at changing the way data probe works

File: server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java (hunk 5 above)
Comment (chibenwa) [inline server/container/cli-integration/src/test/java/org/apache/james/cli/DataCommandsIntegrationTest.java]: Idem

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**